packages/graphql_codegen/test/assets/fragments_and_loose_type_requirements/document.graphql.dart

```diff
@@ -71,6 +71,32 @@ extension UtilityExtension$Fragment$F on Fragment$F {
         this,
         (i) => i,
       );
+  _T when<_T>({required _T Function(Fragment$F$$T) t}) {
+    switch ($__typename) {
+      case "T":
+        return t(this as Fragment$F$$T);
+
+      default:
+        throw Exception("Unknown typename '${$__typename}'");
```

Comment:
I am worried about the backward-compatible aspect of this. Let us say that a new concrete implementation of an interface is added. Then this has the potential to break any app in production.

Is making the return type nullable making more sense? Or providing an `orElse` which can even. be required.

Reply:
went with the `orElse` to address this concern